Sources/StreamChatUI/CommonViews/AvatarView/ChatChannelAvatarView.swift

```diff
@@ -30,31 +47,293 @@ open class ChatChannelAvatarView: _View, ThemeProvider, SwiftUIRepresentable {
                 preferredSize: .avatarThumbnailSize,
                 components: components
             )
+            presenceAvatarView.isOnlineIndicatorVisible = false
             return
         }
 
-        let (avatarURL, isOnlineIndicatorVisible): (URL?, Bool) = {
-            // Try to get the explicit channel avatar first
-            if let avatarURL = channel.imageURL {
-                return (avatarURL, false)
-            }
-
-            // TODO: https://stream-io.atlassian.net/browse/CIS-652
-            // this is just placeholder implementation:
-
-            let firstOtherMember = channel.lastActiveMembers
-                .sorted { $0.memberCreatedAt < $1.memberCreatedAt }
-                .first(where: { $0.id != content.currentUserId })
-
-            return (firstOtherMember?.imageURL, firstOtherMember?.isOnline ?? false)
-        }()
-
+        loadAvatar(for: channel)
+    }
+    
+    open func loadAvatar(for channel: ChatChannel) {
+        // If the channel has an avatar set, load that avatar
```

Comment:
// If the channel has an avatar set, load that avatar
// Use the appropriate method to load avatar based on channel type

The comments are redundant on my opinion, this code is good and readable, so we can remove them but this is up to you.